healthz/README.md

```diff
@@ -7,41 +14,239 @@ in distributed systems.
 The ability to also include "implementation specific" details about the health
 of the system or component is very standard for network devices in the form for
 "show tech", "diag" or "debug" commands. These are very useful for getting
-useful diagnostics about the system state to the implementers. Healthz exposes
+diagnostics about the system state to the implementers. Healthz exposes
 these interfaces as queriable endpoints to allow operators to validate "health"
-on components and if "unhealthy" gather implementor specific data to help triage
-or reproduce issues.
-
-## Design
-
-* The service definition provides a Get method which is correlated with a
-  /component in the OC tree. The Get method is responsible for getting the
-  current status of the requested component path. This will cause the system
-  to fetch the current state of component this may entail running specific
-  validations and gathering any relevant status of the component to reflect
-  the "health check"
-
-* The internal design details of the "health check" and the frequency of the
-  health check are up to the implementor to determine. This presentation of
-  the status in OC is proposed to be modeled as
-
-```yang
-component/healthz/status = Enum
-component/healthz/last-unhealthy = Timestamp
-component/healthz/unhealthy_count = Int64
+on components and if "unhealthy" gather implementor specific data to help
+triage or reproduce issues.
+
+## Design Background
+
+`Healthz` provides a means by which a user may initiate checks of health
+(through the `Check` RPC) or a system may report the results of a check that it
+has initiated of its own accord. 
+
+Following a health check occuring, a caller uses the `List` RPC to discover the
+artifacts that are assoicated with a component and its corresponding
```

Comment:
It seems confusing to use the `artifact` term when we describe the `ComponentStatus`, as the artifact is a message encompassed in the status.
Later down the section we say that we acknowledge artifacts:

```
Once retrieved each artifact can be 'acknowledged' by the client of the RPC.
```

which confuses me, as we ack the `ComponentStatus` (which is also referred as `event` sometimes in this doc).

Shall we rather say `ComponentStatus` when describing the set of health status messages returned by `Get`/`List` RPCs?

Reply:
I agree -- I've tried to standardise that "event" means a `ComponentStatus`. I think that this is clearer, since `ComponentStatus` itself seems obsure from a human understanding. PTAL (will push this change after sending this comment and addressing the others).